neonvm/controllers/virtualmachine_controller.go

```diff
@@ -469,12 +473,24 @@ func (r *VirtualMachineReconciler) doReconcile(ctx context.Context, virtualmachi
 
 			// check if need hotplug/unplug CPU or memory
 			// compare guest spec and count of plugged
-			if virtualmachine.Spec.Guest.CPUs.Use.RoundedUp() != pluggedCPU {
-				log.Info("VM goes into scaling mode, CPU count needs to be changed",
-					"CPUs on board", pluggedCPU,
-					"CPUs in spec", virtualmachine.Spec.Guest.CPUs.Use.RoundedUp())
+
+			specUseCPU := virtualmachine.Spec.Guest.CPUs.Use
+			scaleCgroupCPU := supportsCgroup && *specUseCPU != cgroupUsage.VCPUs
+			scaleQemuCPU := specUseCPU.RoundedUp() != pluggedCPU
+			if scaleCgroupCPU || scaleQemuCPU {
```

Comment:
Definitely I agree it would be nice to make everything simpler and assume all runners support cgroup. However, I don't understand what's annoying here. If `!supportsCgroup` and `specUseCPU.RoundedUp() != pluggedCPU` means we gotta only scale QEMU and we'll take care of that in the `VmScaling` phase.

Reply:
Ah, what I mean is: If a user requests fractional CPU (by setting `spec.guest.cpus.use`) but the VM's current runner pod doesn't support it, what is the "correct" answer here? IMO it's not worth thinking about -- all runners already support fractional CPU.

Reply:
I kept the logic the same as before, so we'll always round up. Anyway, I'll resolve this and we could make things simpler by working on #710 later.